numpy/lib/_function_base_impl.py

```diff
@@ -3827,7 +3827,8 @@ def sinc(x):
 
     """
     x = np.asanyarray(x)
-    y = pi * where(x == 0, 1.0e-20, x)
+    x = x.astype(np.result_type(x.dtype, np.float16))
```

Comment:
Would it make sense to use single precision here?

Admittedly there could be cases where half precision is desirable

Reply:
I checked on main and int8 input produced float16 output.

Reply:
This surprises me – has that changed since the 2.1.3 release?
```python
In [1]: import numpy as np

In [2]: np.__version__
Out[2]: '2.1.3'

In [3]: np.sinc(np.arange(4, dtype='uint8')).dtype
Out[3]: dtype('float64')
```

Reply:
hmm, I may be mistaken

Reply:
I had checked `sin` instead:

```python
In [1]: import numpy as np

In [2]: np.sin(np.arange(4, dtype='uint8')).dtype
Out[2]: dtype('float16')

In [3]: np.sinc(np.arange(4, dtype='uint8')).dtype
Out[3]: dtype('float64')
```